rotkehlchen/chain/ethereum/modules/balancer/decoder.py

```diff
@@ -0,0 +1,291 @@
+            for event in decoded_events:
+                if (
+                    event.asset == A_ETH and event.balance.amount == amount_in_if_eth and
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_type = HistoryEventType.TRADE
+                    event.event_subtype = HistoryEventSubType.SPEND
+                    event.notes = f'Swap {event.balance.amount} {self.eth.symbol} in balancer V2 from {event.location_label}'  # noqa: E501
+                    event.counterparty = CPT_BALANCER_V2
+
+        return None, [action_item]
+
+    def decode_balancer_v2_event(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: list[ActionItem],
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        if tx_log.topics[0] == V2_SWAP:
+            return self._decode_swap_creation(
+                tx_log=tx_log,
+                decoded_events=decoded_events,
+                action_items=action_items,
+            )
+        return None, []
+
+    def _maybe_enrich_balancer_v2_transfers(  # pylint: disable=no-self-use
+            self,
+            token: 'EvmToken',  # pylint: disable=unused-argument
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            event: HistoryBaseEntry,
+            action_items: list[ActionItem],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+    ) -> bool:
+        """
+        Enrich tranfer transactions to address for jar deposits and withdrawals
+        May raise:
+        - UnknownAsset
+        - WrongAssetType
+        """
+        if action_items is None or len(action_items) == 0 or transaction.to_address != VAULT_ADDRESS:  # noqa: E501
+            return False
+
+        assert action_items[-1].extra_data is not None
+        asset = event.asset.resolve_to_evm_token()
+        to_amount = asset_normalized_value(
+            amount=action_items[-1].extra_data['amount_out'],
+            asset=asset,
+        )
+
+        if (
+            action_items[-1].extra_data['to_token'] != tx_log.address or
+            event.balance.amount != to_amount
+        ):
+            return False
+
+        is_receive = asset.evm_address == action_items[-1].extra_data['to_token']
+
+        event.counterparty = CPT_BALANCER_V2
+        event.event_type = HistoryEventType.TRADE
+        if is_receive:
+            event.event_subtype = HistoryEventSubType.RECEIVE
+            event.notes = f'Receive {event.balance.amount} {asset.symbol} in balancer V2 from {event.location_label}'  # noqa: E501
```

Comment:
What exactly is this event? I am trying to understand the notes but I don't grok it.

`Receive 10 GNO in balancer V2 from 0x2b...20`

This sounds wrong. What did you get and from where? The information I need to see is that I got 10 GNO, the address on which I got them and for what reason. So

`Receive 10 GNO in 0x2b..01 from balancer v2 pool XYZ`

Reply:
I have added this information to the message. The balancer tokens have all the same name symbol so is not easy to obtain that information but everything else can be done